Entity/CategoryManager.php

```diff
@@ -143,11 +143,11 @@ public function getRootCategory($context = null)
     /**
      * NEXT_MAJOR: add this method to the interface.
      *
-     * @param ContextInterface $context
+     * @param ContextInterface|string|null $context
      *
      * @return CategoryInterface[]
      */
-    public function getRootCategoriesForContext(ContextInterface $context = null)
+    public function getRootCategoriesForContext($context = null)
```

Comment:
I checked through:
 We stub only one method `CategoryManager::getRootCategoriesSplitByContexts` in `CategoryAdminControllerTest`, and I'm sure that we can change mocking of `CategoryManager` it to a real object (How it's done in `CategoryManagerTest`)